Test workflows merge

JSchoenbachler opened the pull request:
These changes were originally on the `test-workflows` branch, but due to a headache of conflicting files, I created a new branch to facilitate the merge.

JSchoenbachler committed:
Re-commit everything from test-workflows original branch

JSchoenbachler committed:
Remove regex for error checking

JSchoenbachler commented:
@jakejh This is now ready for review. Tests are added and error checking no longer uses regex!

jakejh commented on tests/testthat/setup.R:
```diff
@@ -0,0 +1,46 @@
+# to build the path using Sys.info()['user'] in addition to OS.
+# salmonPartialDir = '/genomes/alias/mm10/salmon_partial_sa_index/default'
+os = Sys.info()['sysname']
+if (os == "Darwin") params$salmon$indexDir = gsub('/home/',
```
What's with the spacing?

JSchoenbachler replied:
Fixed with pull, not sure what happened there?

jakejh commented on tests/testthat/setup.R:
```diff
@@ -0,0 +1,46 @@
+fileColname = 'fastq_fetched'
+multiqcDir = file.path(outputDir, 'multiqc_output')
+
+snapshot = function(xObs, path) {
```
Remove the snapshot function from test-array-utils.R.

jakejh commented on tests/testthat/test-fetch.R:
```diff
@@ -0,0 +1,63 @@
+  originalColumns = c('study_accession', 'sample_accession', 'secondary_sample_accession',
+                      'sample_alias', 'sample_title', 'experiment_accession',
+                      'run_accession', 'fastq_md5', 'fastq_ftp', 'fastq_aspera')
+  metadataControl = metadata[, ..originalColumns]
```
For consistency with other packages, please use suffix `Exp` instead of `Control`.

JSchoenbachler replied:
Done, even on salmon output directory in data directory.

jakejh commented on tests/testthat/test-fetch.R:
```diff
@@ -0,0 +1,63 @@
+  print(result)
+  file1 = paste0("./", strsplit(result$localFilepaths, ';')[[1]][1])
+  file2 = paste0("./", strsplit(result$localFilepaths, ';')[[1]][2])
+  print(file1)
```
No printing in tests.

jakejh commented on tests/testthat/test-fetch.R:
```diff
@@ -0,0 +1,63 @@
+
+  expect_true(file.exists(file1))
+  expect_true(file.exists(file2))
+
```
Clean up line spacing.

jakejh commented on tests/testthat/test-fetch.R:
```diff
@@ -0,0 +1,63 @@
+
+
+  print(result)
+  file1 = paste0("./", strsplit(result$localFilepaths, ';')[[1]][1])
```
No double-quotes unless necessary.

jakejh commented on tests/testthat/test-fetch.R:
```diff
@@ -0,0 +1,63 @@
+
+  print(result)
+  file1 = paste0("./", strsplit(result$localFilepaths, ';')[[1]][1])
+  file2 = paste0("./", strsplit(result$localFilepaths, ';')[[1]][2])
```
Can you not make this and the expect_true into a for loop?

JSchoenbachler replied:
Done.

jakejh commented on tests/testthat/test-seeker.R:
```diff
@@ -0,0 +1,89 @@
+
+test_that('Test checkSeekerArgs', {
+  skip_on_os('windows', arch = NULL)
+  outputDirObs = seeker:::checkSeekerArgs(params, parentDir)
```
No `:::` in tests.

jakejh commented on tests/testthat/test-tximport.R:
```diff
@@ -0,0 +1,44 @@
+
+  tximportControl = snapshot(tximportObs, file.path(dataDir, 'tximport_output.qs'))
+
+  expect_equal(tximportObs$abundance, tximportControl$abundance)
```
Use for loop.

JSchoenbachler committed:
Update per jakes requests on pull request

JSchoenbachler commented:
@jakejh Updated per your feedback.

jakejh commented on tests/testthat/test-salmon.R:
```diff
@@ -0,0 +1,23 @@
+    paramsNow))
+  salmonOutputObs = list.files(salmonDir, recursive = TRUE)
+  salmonOutputExp = snapshot(salmonOutputObs, file.path(dataDir, 'salmon_output.qs'))
+  expect_equal(sort(salmonOutputObs), sort(salmonOutputExp))
```
sort should be unnecessary.

jakejh commented on tests/testthat/test-salmon.R:
```diff
@@ -0,0 +1,23 @@
+test_that('Test salmon and getSalmonMetadata', {
```
If we're only comparing the file lists, I don't want to keep the entire salmon_output directory in the repo.

JSchoenbachler replied:
Are you talking about the `data/salmon_output_exp` directory? That isn't used for this test, it's used for the `tximport` tests. See that script. I think it's necessary to keep.

JSchoenbachler committed:
Remove sort

JSchoenbachler commented:
Fixed one thing, clarified on another.

jakejh commented on tests/testthat/setup.R:
```diff
@@ -0,0 +1,41 @@
+dataDir = 'data'
+params = yaml::read_yaml(file.path(dataDir, 'GSE143524.yml'))
+os = Sys.info()['sysname']
+if (os == 'Darwin') params$salmon$indexDir = gsub('/home/', '/Users/',
```
Still not the preferred style, since indentation is so extreme. Put the code that's inside the `if` on its own line.

jakejh commented on tests/testthat/test-fetch.R:
```diff
@@ -0,0 +1,51 @@
+test_that('Test fetchMetadata', {
```
Do we need to preface every function name with "Test "? I don't think our other packages do that.

jakejh commented on tests/testthat/test-seeker.R:
```diff
@@ -0,0 +1,87 @@
+})
+
+test_that('Test checkSeekerArgs Errors', {
+  skip_on_os('windows', arch = NULL)
```
Can we get some line breaks here?

jakejh reviewed (changes requested):
Ok, I see your point re tximport and salmon_output_exp. Thanks for quick replies.

JSchoenbachler committed:
Fixed spacing

JSchoenbachler committed:
Remove test from context name

JSchoenbachler commented:
@jakejh resolved your issues!

jakejh commented:
How big are the tests/testthat/data folder and the built tar.gz file?

jakejh commented:
I also don't see that you've updated the DESCRIPTION file. Have you looked at the checklist you made in Notion?

JSchoenbachler commented:
22.1 MB on the data folder, the tar.gz is 11.1 MB. I haven't gone all the way through the checklist yet, I will do that!

JSchoenbachler commented:
@jakejh going through the steps for the pull request, I'm just noticing we don't have a NEWS file. should I add one?

JSchoenbachler committed:
Update version number and imports

jakejh commented:
I haven't bothered to make a NEWS file yet, since the package was changing so rapidly. Let's hold off for now.

jakejh commented:
In the 11 MB, what takes up the most space?

JSchoenbachler commented:
Definitely the tests, when I extract the archive, everything other than the test folder combined only comes to a couple hundred KB.

jakejh commented:
I meant what files in particular inside the testthat/data folder.

JSchoenbachler commented:
The `GSE143524` directory is 5.7 MB, `salmon_output_exp` directory is 3.2 MB, `tx2gene.csv` is 5.4 MB, and `E-MTAB-8714_eset.qs` is 4MB.

If we could change `tx2gene.csv` to be a `qs` file, it would go down to about 790KB. We'd just have to modify the `seeker` function to use `qread` instead of `fread`.

jakejh commented:
Where did tx2gene.csv come from? `getTx2Gene()` is supposed to save the file as tx2gene.csv.gz.

JSchoenbachler committed:
Use tx2gene.csv.gz

JSchoenbachler commented:
@jakejh I think I had unzipped for some reason and just kept it that way. It's now using the .gz version and it's much smaller.

JSchoenbachler commented:
I mean doing that though doesn't do anything for the tar.gz since that already compresses everything. It just reduces the size of the data folder on the repository to 12.9 MB.

jakejh commented:
Where does the code use the tx2gene.csv.gz file in the tests/testthat/data folder? There's also one in the GSE143524 folder.

JSchoenbachler committed:
Remove unecessary file

JSchoenbachler commented:
It's not being used, just removed it!

jakejh commented:
Ok, I want you to carefully go over every single line in the test scripts, and ask yourself if this meets our standards. Try to find at least three things to improve.

JSchoenbachler committed:
Various changes based on review

Fixed up formatting and order in setup.R script, changed metadata.csv to metadata.qs to further use qs package, using `names` function for tximport test, change fetchMetadata functions to use snapshot function, fixed where fetch test was downloading to wrong directory, removed same variables being set multiple times

JSchoenbachler commented:
@jakejh Made the following changes-
- Fixed up formatting and order in setup.R script
- Changed metadata.csv to metadata.qs to further use qs package
- Using `names` function for tximport test
- change fetchMetadata functions to use snapshot function
- Fixed where fetch test was downloading to wrong directory
- Removed same variables being set multiple times

jakejh commented:
Excellent, nice job. I just noticed that testthat in Imports is not quite in alphabetical order.

JSchoenbachler committed:
Fix order of imports

JSchoenbachler commented:
@jakejh okay that's fixed now!
